### DISCUSSION TITLE
Please, make a manual. A guide with screenshots, descriptions, related options and links to master all the settings.

### DISCUSSION THREAD

**Discussion by DantesNoWar** [msasanmh/DNSveil#216] [Ideas]
Hi, I'm new. I tried to use SimpleDNSCrypt before but it's abandoned, so went here.
I'm now trying DNSVeil but I can't understand more than half of the settings. It's not much user friendly and settings has no descriptions when hovering the pointer over  the sections.
Could you please make a proper guide that explains everything?
Also, I've tested the connection [here](https://www.dnsleaktest.com/results.html), it seems that servers changes but I ignore the extent of this as a proof of functioning since there's not details of the connection. However, IP doesn't change.
Could you please explain how to set Anonymized DNSCrypt with DNSVeil?
To me it's hard to safely use and trust a piece of software I don't understand much.
If there's a documentation I missed, could you please share it or tell me where to find it? Unfortunately, I didn't found it by my own.
Thanks for your time.
Dante

**Comment by msasanmh** [comment]
> Hi, I'm new. I tried to use SimpleDNSCrypt before but it's abandoned, so went here.

Hi

> I'm now trying DNSVeil but I can't understand more than half of the settings. It's not much user friendly and settings has no descriptions when hovering the pointer over the sections.

I know. I've been working on the Core (Library) of this app to support almost everything about DNS and Proxy. I couldn't find enough time to write the documentation, also writing the docs is a nightmare to any programmer.

> Could you please make a proper guide that explains everything?

The few next version would be about fixing bugs on this current app (SDC - SecureDNSClient).
This current platform has some limitations hence it's getting my freedom.
I'm writing the whole app from scratch on a newer platform, then I can add anything to it including explanations for settings. 

> Also, I've tested the connection [here](https://www.dnsleaktest.com/results.html), it seems that servers changes but I ignore the extent of this as a proof of functioning since there's not details of the connection.

When you are using the built-in servers, the app checks servers on the background, it replaces slow servers with faster ones according to your `Check` settings.
When you are using your custom servers, the app checks your custom servers on the background to see if they work, it replaces non working servers with online servers, if all your servers are offline, app will replace them with built-in servers (since built-in servers are secure and uncensored). this behavior will change in the new platform.

> However, IP doesn't change.

A DNS client never changes your IP address, only proxies and VPN services change your IP address.

> Could you please explain how to set Anonymized DNSCrypt with DNSVeil?

You need to use Custom Servers.
A combination of a DNSCrypt Target and DNSCrypt Relay is an Anonymized DNSCrypt.
You should write it like this:
`DNSCrypt Target` `space` `DNSCrypt Relay`

you can find DNSCrypt servers from this page:
https://dnscrypt.info/public-servers

and you can find DNSCrypt Relay from:
https://github.com/DNSCrypt/dnscrypt-resolvers/blob/master/v3/relays.md

for example these servers are Anonymized DNSCrypt:
```
sdns://AQcAAAAAAAAADjk1LjEzMS4yMDIuMTA1ICNqYnU4LMuHNFVgCP5Zn1414WbRxXWqmbQoFp-KjKepGTIuZG5zY3J5cHQtY2VydC5kbnNjcnkucHQ sdns://gQ4zNy4xMjAuMTQyLjExNQ
```
```
sdns://AQcAAAAAAAAADjE3My4yNDkuMjAzLjUyIC7cQyu7dTkvrH-MdwA3fzAkI_2dEUnZd3x1ON0tfYubGTIuZG5zY3J5cHQtY2VydC5kbnNjcnkucHQ sdns://gQ0zNy4xMjAuMjE3Ljc1
```
```
sdns://AQcAAAAAAAAACzg5LjExNy4yLjE3IAXdC7hGEegKD86br-tVRwZTcJfJZAEFjW4jCV5lzdutGTIuZG5zY3J5cHQtY2VydC5kbnNjcnkucHQ sdns://gQ8xOTAuMjExLjI1NS4yMjc
```

So, on `Check` tab right click on `Custom Servers`
Select `Manage custom servers`
Click `New group...` and give a name to it.
Write the servers exactly like the above example.
 
![111](https://github.com/user-attachments/assets/0892b401-e065-47db-ab64-6195a92e3cf5)

Close `Manage custom servers` window.
On `Check` tab select `Use custom servers`.
Right click on `Scan` and select your group name.

Your log window would be like this:
![222](https://github.com/user-attachments/assets/f0d90f67-ebfe-4067-a0e7-6ab42c3f2ffe)

Go to `Connect` tab and select `Connect to working servers` and click on `Connect` to connect to your scanned servers.
Go to `Set DNS` tab and set DNS on your Network Adapter.

Done. Now you are connected to your Anonymized DNSCrypt servers and the whole system is using it.
You can do all these above steps with one click if you configure the Quick Connect settings (Settings -> Quick Connect).

Once you configured Quick Connect you can use `Quick Connect` button on `Check` tab or by right clicking on the app tray icon.

> To me it's hard to safely use and trust a piece of software I don't understand much.

This app is fully open source, everything is transparent.

> If there's a documentation I missed, could you please share it or tell me where to find it? Unfortunately, I didn't found it by my own. Thanks for your time. Dante

The only documentation is one the main page for now:
https://github.com/msasanmh/DNSveil#notes
